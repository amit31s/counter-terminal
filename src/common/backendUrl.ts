/* istanbul ignore file */

import { featureFlags } from "@ct/constants";
import isElectron from "is-electron";
import { envProvider } from "./platformHelper";

const forgeRockRedirectUrl =
  envProvider.NODE_ENV === "development" || isElectron()
    ? envProvider.REACT_APP_FORGEROCK_COGNITO_REDIRECT_ELECTRON
    : envProvider.REACT_APP_FORGEROCK_COGNITO_REDIRECT_SIM;

// should contain backend urls for different environments to make switching easier
export const SERVER_ROOT: string = `https://api.spm-${envProvider.REACT_APP_ENV}.com` || "";

export const AWS_COGNITO_REGION: string = envProvider.REACT_APP_AWS_COGNITO_REGION || "";
export const AWS_COGNITO_USER_POOL_ID: string =
  envProvider.REACT_APP_AWS_COGNITO_USER_POOL_ID || "";
export const AWS_COGNITO_WEB_CLIENT_ID: string =
  envProvider.REACT_APP_AWS_COGNITO_WEB_CLIENT_ID || "";

export const FORGEROCK_COGNITO_REGION: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_REGION || "";
export const FORGEROCK_COGNITO_USER_POOL_ID: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_USER_POOL_ID || "";
export const FORGEROCK_COGNITO_WEB_CLIENT_ID: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_WEB_CLIENT_ID || "";
export const FORGEROCK_COGNITO_OAUTH_DOMAIN: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_OAUTH_DOMAIN || "";
export const FORGEROCK_COGNITO_S3_BUCKET: string =
  envProvider.REACT_APP_FORGEROCK_COGNITO_S3_BUCKET || "";
export const FORGEROCK_COGNITO_REDIRECT: string = forgeRockRedirectUrl || "";

const federationManagerUrlLocal = "http://localhost:4000/apps/federation-manager";
const federationManagerUrlDeployed =
  (envProvider.REACT_APP_USING_ELECTRON === "true"
    ? envProvider.REACT_APP_FEDERATION_MANAGER_URL_STANDALONE
    : envProvider.REACT_APP_FEDERATION_MANAGER_URL_SIMULATOR) || "";

export const FEDERATION_MANAGER_URL: string = featureFlags?.useLocalFederationManager
  ? federationManagerUrlLocal
  : federationManagerUrlDeployed;

export const POL_DEVICE_SERVER_HOST: string = envProvider.REACT_APP_POL_DEVICE_SERVER_HOST || "";
export const POL_DEVICE_SERVER_SIMULATED: boolean =
  envProvider.REACT_APP_USING_ELECTRON === "true"
    ? envProvider.REACT_APP_POL_DEVICE_SERVER_SIMULATED_ELECTRON === "true"
    : envProvider.REACT_APP_POL_DEVICE_SERVER_SIMULATED_SIMULATOR === "true";
export const CASH_TRANSFER_OUT_DISABLED: boolean =
  envProvider.REACT_APP_CASH_TRANSFER_OUT_DIASBLED === "true" || false;
export const POUCH_ACCEPTANCE_DISABLED: boolean =
  envProvider.REACT_APP_POUCH_ACCEPTANCE_DISABLED === "true" || false;
export const LOGOUT_CASHDRAWER_DISABLED: boolean =
  envProvider.REACT_APP_LOGOUT_CASHDRAWER_DISABLED === "true" || false;
export const POUCH_DISPATCH_DISABLED: boolean =
  envProvider.REACT_APP_POUCH_DISPATCH_DISABLED === "true" || false;
export const PMS_VISIBLE: boolean = envProvider.REACT_APP_PMS_VISIBLE === "true" || false;
export const QUANTITY_BUTTON_DISABLED: boolean =
  envProvider.REACT_APP_QUANTITY_BUTTON_DISABLED === "true" || false;
export const INITIALISE_PED_ON_LOGIN: boolean =
  envProvider.REACT_APP_INITIALISE_PED_ON_LOGIN === "true" || false;
// allows for local override of tpv for pin pad
export const POL_PED_TPV_OVERRIDE: string = envProvider.REACT_APP_POL_PED_TPV_OVERRIDE || "";
export const MSR_ENABLED: boolean = envProvider.REACT_APP_MSR_ENABLED === "true" || false;
export const REACT_APP_FORGEROCK_COGNITO_WEB_CLIENT_ID =
  envProvider.REACT_APP_FORGEROCK_COGNITO_WEB_CLIENT_ID;
