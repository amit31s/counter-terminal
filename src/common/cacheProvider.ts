import Cache from "@aws-amplify/cache";
import { CacheItemOptions as AmplifyCacheInterface } from "@aws-amplify/cache/src/types/Cache";

Cache.configure({
  itemMaxSize: 400000, // 400kb
});

type CacheValue = string | number | boolean | Record<string, unknown> | unknown[];

const setItem = (key: string, value: CacheValue, options?: AmplifyCacheInterface): void =>
  Cache.setItem(key, value, options);
const getItem = (key: string, options?: AmplifyCacheInterface): CacheValue | unknown =>
  Cache.getItem(key, options);
const removeItem = (key: string): void => Cache.removeItem(key);
const clear = (): void => Cache.clear();

export default Object.freeze({
  setItem,
  getItem,
  removeItem,
  clear,
});
