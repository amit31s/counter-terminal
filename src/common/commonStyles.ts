import { StyleSheet } from "react-native";
import { scale } from "../utils/Scaling/ScalingUtility";
import colorConstants from "../constants/ColorConstants";
export default StyleSheet.create({
  darkHeader: {
    width: "100%",
    height: scale(32),
    backgroundColor: colorConstants.primaryDark,
  },
  headerContainer: {
    width: "100%",
    height: scale(52),
    backgroundColor: colorConstants.primaryColour,
  },
  header: {
    fontSize: 20,
    fontWeight: "normal",
    marginLeft: scale(40),
    marginTop: scale(10),
    color: colorConstants.white,
  },
});
