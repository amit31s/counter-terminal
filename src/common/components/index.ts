export * from "./CurrencyInput";
export * from "./CustomButton";
export * from "./ErrorBoundary";
export * from "./PressableText";
export * from "./SalesReceiptModal";
export * from "./ScannerInput";
export * from "./Screen";
export * from "./ScreenHeader";
export * from "./ScreenLayout";
export * from "./SpringPressable";
