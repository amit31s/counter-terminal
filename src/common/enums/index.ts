export * from "./errorEnum";
export * from "./storageKeys";
