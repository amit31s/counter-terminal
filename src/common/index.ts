export * from "./backendUrl";
export * from "./backOfficeUrl";
export * from "./components";
export * from "./getProduct";
export * from "./helpers";
export * from "./hooks";
export * from "./selectors";
export * from "./state";
export * from "./theme";
