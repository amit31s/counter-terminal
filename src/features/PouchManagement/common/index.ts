export * from "./SubmitPouchButton";
export * from "./UnableToScan";
