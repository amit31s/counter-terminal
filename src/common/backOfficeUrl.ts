import { STORAGE_KEYS } from "@ct/common/enums";
import { getItem } from "@ct/utils";
import { envProvider } from "./platformHelper";

export enum BackOfficePageEnum {
  taskManagement = "taskManagement",
  pouchPreperation = "pouchPreperation",
}

export const generateBackOfficeURL = () => {
  const currentPage = getItem(STORAGE_KEYS.CTSTK0003);
  const deviceFadcode = getItem(STORAGE_KEYS.CTSTK0004);
  return `${envProvider.REACT_APP_BACK_OFFICE_URL}/${currentPage}?navigatedFromCT=true&fadcode=${deviceFadcode}`;
};
