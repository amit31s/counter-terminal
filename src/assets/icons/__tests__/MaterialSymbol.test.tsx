import { render } from "@ct/common";
import { MaterialSymbol } from "..";
const SIZES = ["small", "medium", "large"] as const;
const COLORS = [undefined, "pink", "#FFF000"] as const;

describe("Material Symbol", () => {
  it("renders with the expected styles", () => {
    const view = render(
      <>
        {SIZES.map((size) =>
          COLORS.map((color) => <MaterialSymbol name="home" size={size} color={color} />),
        )}
      </>,
    );
    expect(view.baseElement).toMatchSnapshot();
  });
});
