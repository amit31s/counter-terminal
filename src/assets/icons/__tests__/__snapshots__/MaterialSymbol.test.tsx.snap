// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Material Symbol renders with the expected styles 1`] = `
.emotion-1 {
  font-variation-settings: 'FILL' 1,'wght' 400,'GRAD' 0,'opsz' 24;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 24px;
  width: 24px;
  font-size: 24px;
  color: white;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-2 {
  font-variation-settings: 'FILL' 1,'wght' 400,'GRAD' 0,'opsz' 24;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 24px;
  width: 24px;
  font-size: 24px;
  color: pink;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-3 {
  font-variation-settings: 'FILL' 1,'wght' 400,'GRAD' 0,'opsz' 24;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 24px;
  width: 24px;
  font-size: 24px;
  color: #FFF000;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-4 {
  font-variation-settings: 'FILL' 0,'wght' 400,'GRAD' 0,'opsz' 48;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 48px;
  width: 48px;
  font-size: 48px;
  color: white;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-5 {
  font-variation-settings: 'FILL' 0,'wght' 400,'GRAD' 0,'opsz' 48;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 48px;
  width: 48px;
  font-size: 48px;
  color: pink;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-6 {
  font-variation-settings: 'FILL' 0,'wght' 400,'GRAD' 0,'opsz' 48;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 48px;
  width: 48px;
  font-size: 48px;
  color: #FFF000;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-7 {
  font-variation-settings: 'FILL' 0,'wght' 400,'GRAD' 0,'opsz' 60;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 60px;
  width: 60px;
  font-size: 60px;
  color: white;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-8 {
  font-variation-settings: 'FILL' 0,'wght' 400,'GRAD' 0,'opsz' 60;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 60px;
  width: 60px;
  font-size: 60px;
  color: pink;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-9 {
  font-variation-settings: 'FILL' 0,'wght' 400,'GRAD' 0,'opsz' 60;
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 60px;
  width: 60px;
  font-size: 60px;
  color: #FFF000;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

<body>
  <div>
    <div
      class="emotion-0 r-flex-13awgt0"
    >
      <span
        class="material-symbols-outlined emotion-1"
      >
        home
      </span>
      <span
        class="material-symbols-outlined emotion-2"
      >
        home
      </span>
      <span
        class="material-symbols-outlined emotion-3"
      >
        home
      </span>
      <span
        class="material-symbols-outlined emotion-4"
      >
        home
      </span>
      <span
        class="material-symbols-outlined emotion-5"
      >
        home
      </span>
      <span
        class="material-symbols-outlined emotion-6"
      >
        home
      </span>
      <span
        class="material-symbols-outlined emotion-7"
      >
        home
      </span>
      <span
        class="material-symbols-outlined emotion-8"
      >
        home
      </span>
      <span
        class="material-symbols-outlined emotion-9"
      >
        home
      </span>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;
